server/pulp/server/db/migrations/0029_applicability_schema_change.py

```diff
@@ -51,7 +51,7 @@ def migrate(*args, **kwargs):
     try:
         rpa_collection.drop_index("profile_hash_-1_repo_id_-1")
     except pymongo.errors.OperationFailure as e:
-        if e.code == 27:
+        if not e.code or e.code == 27:
             # index not found - good, it's been removed before
             pass
         else:
```

Comment:
I think that in case the `e.code` being `None` isn't unique to the index not found case, this might never be executed on the older versions of Mongo, that don't populate the `e.code`, silencing other issues.

Reply:
@dparalen , I agree completely but we need it to work on old Mongo, in my opinion, until we are ready to require mongo 3